src/NHibernate/Transaction/AdoNetWithSystemTransactionFactory.cs

```diff
@@ -285,16 +285,17 @@ protected virtual void Unlock()
 				{
 					// Cloned transaction is not disposed "unexpectedly", its status is accessible till context disposal.
 					var status = EnlistedTransaction.TransactionInformation.Status;
-					if (status != TransactionStatus.Active)
+					if (status != TransactionStatus.Active || _preparing)
```

Comment:
The trouble with dependent transactions occurs in the prepare phase, where we are sure the transaction is not (yet) roll-backed.
This way of dodging that trouble looks like one more crutch in this dreaded system transaction management. But I have seen no better.

(Checking the original transaction concrete type is unreliable, because a `DependentTransaction` own `Clone` method will not yield another `DependentTransaction`, but a `Transaction`. Moreover, knowing it is a dependent transaction will anyway not help knowing whether the transaction is still actually active or not.)